projects/storefrontlib/src/cms-components/asm/asm-root/asm-root.component.spec.ts

```diff
@@ -76,4 +78,40 @@ describe('AsmRootComponent', () => {
     fixture.detectChanges();
     expect(el.query(By.css('cx-asm-main-ui'))).toBeFalsy();
   });
+
+  describe('Minimize ASM UI', () => {
+    it('should expand the ASM UI when the expand arrow button is clicked', () => {
+      spyOn(asmService, 'getAsmUiState').and.returnValue(
+        of({ visible: true, expanded: false })
+      );
+      spyOn(asmService, 'updateAsmUiState').and.stub();
+
+      component.ngOnInit();
+      fixture.detectChanges();
+
+      const expandBtn = fixture.debugElement.query(By.css('button'));
+
+      expandBtn.nativeElement.click();
+      expect(asmService.updateAsmUiState).toHaveBeenCalledWith({
+        expanded: true,
+      });
+    });
+
+    it('should collapse the ASM UI when the collapse button is clicked', () => {
+      spyOn(asmService, 'getAsmUiState').and.returnValue(
+        of({ visible: true, expanded: true })
+      );
+      spyOn(asmService, 'updateAsmUiState').and.stub();
+
+      component.ngOnInit();
+      fixture.detectChanges();
+
+      const collapseBtn = fixture.debugElement.query(By.css('button'));
+
+      collapseBtn.nativeElement.click();
+      expect(asmService.updateAsmUiState).toHaveBeenCalledWith({
+        expanded: false,
+      });
+    });
+  });
 });
```

Comment:
Can we improve coverage with 2 tests:
- Element cx-asm-main-ui should have the collapsed class when state is: expanded = false
- Element cx-asm-main-ui should not have the collapsed class when state is: expanded = true